L1Trigger/DemonstratorTools/plugins/GTTOutputFileReader.cc

```diff
@@ -0,0 +1,176 @@
+
+ Description: Example EDProducer class, illustrating how BoardDataReader can be used to
+   read I/O buffer files (that have been created in hardware/firmware tests), decode
+   the contained data, and store this in EDM collections.
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Thomas Williams <thomas.williams@stfc.ac.uk>
+//         Created:  Fri, 19 Feb 2021 01:10:55 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/L1Trigger/interface/Vertex.h"
+#include "L1Trigger/DemonstratorTools/interface/BoardDataReader.h"
+#include "L1Trigger/DemonstratorTools/interface/codecs/vertices.h"
+#include "L1Trigger/DemonstratorTools/interface/utilities.h"
+
+//
+// class declaration
+//
+
+class GTTOutputFileReader : public edm::stream::EDProducer<> {
+public:
+  explicit GTTOutputFileReader(const edm::ParameterSet&);
+  ~GTTOutputFileReader() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void beginStream(edm::StreamID) override;
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endStream() override;
+
+  //virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  //virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+  //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+  //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+
+  // ----------member data ---------------------------
+  l1t::demo::BoardDataReader fileReader_;
+};
+
+//
+// constants, enums and typedefs
+//
+
+// NOTE: At least some of the info from these constants will eventually come from config files
+constexpr size_t kGapLength(44);
+constexpr size_t kVertexTMUX(6);
+constexpr size_t kVertexChanIndex(0);
+
+const std::map<size_t, l1t::demo::ChannelSpec> kChannelSpecs = {
+    /* channel index -> {link TMUX, TMUX index, inter-packet gap} */
+    {kVertexChanIndex, {kVertexTMUX, 0, kGapLength}}};
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+
+GTTOutputFileReader::GTTOutputFileReader(const edm::ParameterSet& iConfig)
+    : fileReader_(l1t::demo::parseFileFormat(iConfig.getUntrackedParameter<std::string>("format")),
+                  iConfig.getParameter<std::vector<std::string>>("files"),
+                  9,
+                  6,
+                  10,
+                  kChannelSpecs) {
+  produces<l1t::VertexCollection>();
```

Comment:
Wouldn't it be better to create an `l1t::VertexWordCollection` as opposed to an `let::VertexCollection`? Since this information is already digitized and in the word format it doesn't really make sense to store less information in a floating point representation.

Reply:
Yup, absolutely, the word collection is best here

Reply:
Done.